pkg/server/chaosd/jvm_attack.go

```diff
@@ -0,0 +1,233 @@
+				log.Error("failed to update experiment", zap.Error(err))
+			}
+			return
+		}
+
+		// use the stressngPid as recover command, and will kill the pid when recover
+		if err := s.exp.Update(context.Background(), uid, core.Success, "", attack.String()); err != nil {
+			log.Error("failed to update experiment", zap.Error(err))
+		}
+	}()
+
+	if len(attack.Do) == 0 {
+		switch attack.Action {
+		case core.JVMLatencyAction:
+			attack.Do = fmt.Sprintf("Thread.sleep(%s)", attack.LatencyDuration)
+		case core.JVMExceptionAction:
+			attack.Do = fmt.Sprintf("throw new %s", attack.ThrowException)
+		case core.JVMReturnAction:
+			attack.Do = fmt.Sprintf("return %s", attack.ReturnValue)
+		case core.JVMStressAction:
+			if attack.CPUCount > 0 {
+				attack.StressType = "CPU"
+				attack.StressValueName = "CPUCOUNT"
+				attack.StressValue = attack.CPUCount
+			} else {
+				attack.StressType = "MEMORY"
+				attack.StressValueName = "MEMORYSIZE"
+				attack.StressValue = attack.MemorySize
+			}
+
+		}
+	}
+
+	buf := new(bytes.Buffer)
+	if attack.Action != core.JVMStressAction {
+		t := template.Must(template.New("byteman rule").Parse(ruleTemplate))
+		err = t.Execute(buf, attack)
+
+	} else {
+		t := template.Must(template.New("byteman rule").Parse(stressRuleTemplate))
+		err = t.Execute(buf, attack)
+	}
+
+	if err != nil {
+		log.Error("executing template", zap.Error(err))
+		return "", err
+	}
+
+	log.Info("byteman rule", zap.String("rule", string(buf.Bytes())))
+
+	tmpfile, err := ioutil.TempFile("", "rule.btm")
+	if err != nil {
+		return "", err
+	}
+
+	log.Info("create btm file", zap.String("file", tmpfile.Name()))
+
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	if _, err := tmpfile.Write(buf.Bytes()); err != nil {
+		return "", err
+	}
+
+	if err := tmpfile.Close(); err != nil {
+		return "", err
+	}
+
+	bmSubmitCmd := fmt.Sprintf(bmSubmitCommand, attack.Port, "l", tmpfile.Name())
+	cmd := exec.Command("bash", "-c", bmSubmitCmd)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Error(string(output), zap.Error(err))
+		return "", err
+	}
+
+	log.Info(string(output))
+	return uid, nil
+}
+
+func (s *Server) RecoverJVMAttack(uid string, attack *core.JVMCommand) error {
+	// Create a new template and parse the letter into it.
+	t := template.Must(template.New("byteman rule").Parse(ruleTemplate))
+
+	buf := new(bytes.Buffer)
+	err := t.Execute(buf, attack)
+	if err != nil {
+		log.Error("executing template", zap.Error(err))
+		return err
+	}
+
+	tmpfile, err := ioutil.TempFile("", "rule.btm")
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	if _, err := tmpfile.Write(buf.Bytes()); err != nil {
+		return err
+	}
+
+	if err := tmpfile.Close(); err != nil {
+		return err
+	}
+
+	log.Info("create btm file", zap.String("file", tmpfile.Name()))
+
+	bmSubmitCmd := fmt.Sprintf(bmSubmitCommand, attack.Port, "u", tmpfile.Name())
+	cmd := exec.Command("bash", "-c", bmSubmitCmd)
```

Comment:
I know `sh` is better than `bash` now.[Difference between sh and bash](https://stackoverflow.com/questions/5725296/difference-between-sh-and-bash)

Reply:
but in chaos mesh and chaosd, we all use bash now, maybe we can change them all later